userspace/falco/stats_writer.cpp

```diff
@@ -60,7 +60,10 @@ bool stats_writer::init_ticker(uint32_t interval_msec, std::string &err)
 	sev.sigev_value.sival_ptr = &s_timerid;
 #ifndef __EMSCRIPTEN__
 	// delete any previously set timer
-	timer_delete(s_timerid);
+	if (s_timerid)
```

Comment:
Reading here https://bugs.launchpad.net/ubuntu/+source/glibc/+bug/1940296 the bug also occurs when `s_timerid` is something random, like in this case:
```c
int main(void)
{
        timer_t t = (timer_t)0xe236f38802c65008ULL;

        return timer_delete(t);
}
```
BTW i'm not sure this could happen in our case :thinking:

Reply:
Nope because it is either set to `s_timerid = nullptr;` or it is recreated immediately (in this method).

Reply:
Moreover since also `timer_settime` seems to be affected I would do something like
```cpp
if (!s_timerid || timer_settime(s_timerid, 0, &timer, NULL) == -1) {
  ...
}
```
at line 77, WDYT? It's also true that we should fail in `timer_create` if `s_timerid` is invalid

Reply:
yeah, I hope that at least timer_create should fail if `s_timerid` is invalid...

Reply:
Yep i think so :pray:

Reply:
I added a double check that `timer_delete` does not fail. We should be as strength as possible now.

Reply:
ty!

Reply:
Hey @FedeDP I tried this but even after calling timer_create(), it would still be a nullptr and we woudl effectively never delete it ...

Reply:
Whoa; fact is, i can't test this patch myself because my glibc version isn't affected by the bug... this means that this particular glibc version that is exposing the bug is completely utterly broken :/